File: app/src/main/java/com/example/pam_app/adapter/BindingAdapter.java
package com.example.pam_app.adapter;

import androidx.recyclerview.widget.RecyclerView;

public class BindingAdapter {
    @androidx.databinding.BindingAdapter("data")
    public static <T> void setRecyclerViewProperties(RecyclerView recyclerView, T data) {
        if (recyclerView.getAdapter() instanceof BindableAdapter) {
            ((BindableAdapter) recyclerView.getAdapter()).setData(data);
        }
    }
}

File: app/src/main/java/com/example/pam_app/adapter/BindingAdapter.kt
package com.example.pam_app.adapter

import androidx.databinding.BindingAdapter
import androidx.recyclerview.widget.RecyclerView

@Suppress("UNCHECKED_CAST")
object BindingAdapter {
    @kotlin.jvm.JvmStatic
    @BindingAdapter("data")
    fun <T> setRecyclerViewProperties(recyclerView: RecyclerView, items: List<T>?) {
        if (recyclerView.adapter is BindableAdapter<*>) {
            (recyclerView.adapter as BindableAdapter<Collection<T>>?)!!.setData(items)
        }
    }
}